ui/html-fragment.reel/html-fragment.js

```diff
@@ -0,0 +1,217 @@
+                                        attribute,
+                                        child
+                                    );
+
+                                    if (typeof delegateResponse === 'boolean') {
+                                        shouldRemoveAttribute = delegateResponse;
+                                    }
+
+                                    if (shouldRemoveAttribute) {
+                                        child.removeAttribute(attributeName);
+                                        ll--;
+                                        ii--;
+                                    }
+                                } else {
+                                    delegateResponse = this.callDelegateMethod(
+                                        'htmlFragmentWillUseAttributeValueForAttributeNameFromNode',
+                                        this,
+                                        attributeValue,
+                                        attributeName,
+                                        child
+                                    );
+
+                                    if (typeof delegateResponse === 'string') {
+                                        attributeValue = delegateResponse;
+                                    }
+
+                                    child.setAttribute(attributeName, attributeValue);
+                                }
+                            }
+
+                            this._sanitizeNode(child, allowedTags, allowedAttributes);
+                        }
+                    }
+                }                
+            }
+
+            return parent;
+        }
+    },
+
+    _sanitizeHtml: {
+        value: function (html, allowedTags, allowedAttributes) {
+            var doc;
+
+            if (window.DOMParser) {
+                try {
+                    doc = new DOMParser().parseFromString(html, "text/html");
+                } catch (DOMParserError) {
+                    console.error(DOMParserError);
+                }
+
+                if (doc) {
+                    this._sanitizeNode(
+                        doc.body,
+                        allowedTags,
+                        allowedAttributes
+                    );
+
+                    this._sanitizeNode(
+                        doc.head,
+                        allowedTags,
+                        allowedAttributes
+                    );
+                }
+            }
+
+            return doc;
+        }
+    },
+
+    draw: {
+        value: function () {
+            if (this.value && this.needsSanitizeHtml) {
+                var doc = this._sanitizeHtml(
+                    this.value,
+                    this.allowedTags || defaultOptions.allowedTags,
+                    this.allowedAttributes || defaultOptions.allowedAttributes
+                );
+
+                if (doc) {
+                    var range = doc.createRange();
```

Comment:
Same here, I think we can share a Range object among instances since it's not recursive and Range's state is reset here everytime it's used